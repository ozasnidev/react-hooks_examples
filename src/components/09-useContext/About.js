import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const About = () => {
    const {user, setUser } = useContext(UserContext);
    const handleReset = () => {
        setUser({});
    }
    return (
        <div>
            <h1>About</h1>
            <hr />
            <pre>{ JSON.stringify(user, null, 3) }</pre>
            <button
                className="btn btn-danger"
                onClick = { handleReset }
            >
                Logout
            </button>
        </div>
    )
}
