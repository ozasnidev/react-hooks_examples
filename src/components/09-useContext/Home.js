import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const Home = () => {
    const {user} = useContext(UserContext);
    return (
        <div>
            <h1>Home</h1>
            <hr />
            <pre>{JSON.stringify(user, null, 3)}</pre>
        </div>
    )
}
