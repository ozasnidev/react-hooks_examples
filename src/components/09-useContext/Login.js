import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const Login = () => {
    const {setUser} = useContext(UserContext);
    return (
        <div>
            <h1>Login</h1>
            <hr />
            <button 
                className="btn btn-primary"
                onClick={ () => setUser({
                    id: 7,
                    name: 'Juan'
                })}
            >
                Login
            </button>
        </div>
    )
}
