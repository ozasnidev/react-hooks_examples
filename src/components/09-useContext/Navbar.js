import React from 'react'
import { NavLink } from 'react-router-dom'

export const Navbar = () => {
    return (
        <nav className="navbar navbar-expand-lg navbar-light bg-primary">
            <h3 className="navbar-brand">useContext</h3>
            
            <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                <div className="navbar-nav">
                    <NavLink exact activeClassName="active" className="nav-item nav-link" to="/">Home</NavLink>
                    <NavLink exact activeClassName="active" className="nav-item nav-link" to="/login">Login</NavLink>
                    <NavLink exact activeClassName="active" className="nav-item nav-link" to="/about">About</NavLink>
                </div>
            </div>
        </nav>
    )
}
