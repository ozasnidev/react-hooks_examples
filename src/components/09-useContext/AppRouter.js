import React from 'react';
import { Navbar } from './Navbar';
import { About } from './About';
import { Login } from './Login';
import { Home } from './Home';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
  } from "react-router-dom";

export const AppRouter = () => {
    return (
        <Router >
            <>
                <Navbar />
                <Switch>
                    <Route exact path="/" component={ Home } />
                    <Route exact path="/login" component={ Login } />
                    <Route exact path="/about" component={ About } />
                    <Redirect to="/" />
                </Switch>
            </>
        </Router>
    )
}
