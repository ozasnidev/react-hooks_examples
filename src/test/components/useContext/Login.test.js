import React from 'react';
import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import { Login } from '../../../components/09-useContext/Login';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Login component test', () => {
    const setUser = jest.fn();
    const wrapper = mount(
        <UserContext.Provider value={{ setUser }}>
            <Login />
        </UserContext.Provider>
            
    );
        
    test('render', () => {
        expect(wrapper).toMatchSnapshot();
    });
    test('setUser function call', () => {
        wrapper.find('button').simulate('click');
        expect(setUser).toHaveBeenCalledWith({ id: 7, name: 'Juan' });
    });
})
