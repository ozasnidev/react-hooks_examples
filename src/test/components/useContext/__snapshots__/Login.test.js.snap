// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component test render 1`] = `
<div>
  <h1>
    Login
  </h1>
  <hr />
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Login
  </button>
</div>
`;
